### PR TITLE
Share vendor handling

### PR DESCRIPTION
**Pull request by skitt**
For Go-based projects with a go.mod file, this defines a
modules/vendor.txt target which can be used as a dependency to ensure
that the vendor directory is up-to-date.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

File: Makefile.inc [hunk 1]
```diff
@@ -13,3 +13,9 @@ clusters:
 
 deploy: clusters
 	$(SCRIPTS_DIR)/deploy.sh --globalnet $(globalnet) --deploytool $(deploytool) $(DEPLOY_ARGS)
+
+ifeq (go.mod,$(wildcard go.mod))
```
Comment (mkolesnik) [inline Makefile.inc:17]: Why wildcard here? It's only for expansion of patterns AFAICT whilst `go.mod` isn't one.
Also what's the point of this comparison?

Comment (skitt) [reply Makefile.inc:17]: It’s an existence check: `$(wildcard go.mod)` expands to `go.mod` if the file exists, empty otherwise.

Comment (mkolesnik) [reply Makefile.inc:17]: ah i see a bit confusing but cool :)
maybe a comment would've been nice here ;)

Comment (skitt) [reply Makefile.inc:17]: Done!

**Pull request merged**